File: app/src/main/java/com/afoxxvi/alopex/dialog/BaseDialog.java
package com.afoxxvi.alopex.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;

import com.afoxxvi.alopex.R;
import com.afoxxvi.alopex.databinding.DialogBaseBinding;

public abstract class BaseDialog {
    protected Dialog dialog;
    protected final Context context;
    protected final View view;
    protected final DialogBaseBinding baseBinding;

    public BaseDialog(Context context, String title) {
        this.context = context;
        //view = LayoutInflater.from(context).inflate(layout, null);
        baseBinding = DialogBaseBinding.inflate(LayoutInflater.from(context));
        view = baseBinding.getRoot();

        baseBinding.textTitle.setText(title);
        baseBinding.buttonConfirm.setOnClickListener(v -> onConfirm());
        baseBinding.buttonCancel.setOnClickListener(v -> onCancel());
    }

    protected BaseDialog setButtonText(String confirm, String cancel) {
        baseBinding.buttonCancel.setText(cancel);
        baseBinding.buttonConfirm.setText(confirm);
        return this;
    }

    protected BaseDialog setBottomVisible(boolean visible) {
        baseBinding.dialogBottom.setVisibility(visible ? View.VISIBLE : View.GONE);
        return this;
    }

    protected BaseDialog setContent(View view) {
        baseBinding.contentLayout.addView(view);
        return this;
    }

    public void show() {
        dialog = new AlertDialog.Builder(context)
                .setView(view)
                .create();
        Window window = dialog.getWindow();
        window.setBackgroundDrawableResource(R.color.Transparent);
        dialog.show();
    }

    public void onConfirm() {
        if (dialog != null) {
            dialog.dismiss();
        }
    }

    public void onCancel() {
        if (dialog != null) {
            dialog.dismiss();
        }
    }
}

File: app/src/main/java/com/afoxxvi/alopex/ui/dialog/BaseDialog.kt
package com.afoxxvi.alopex.ui.dialog

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import com.afoxxvi.alopex.R
import com.afoxxvi.alopex.databinding.DialogBaseBinding

abstract class BaseDialog(protected val context: Context, title: String?) {
    @JvmField
    protected var dialog: Dialog? = null
    protected val view: View
    private val baseBinding: DialogBaseBinding = DialogBaseBinding.inflate(LayoutInflater.from(context))

    init {
        view = baseBinding.root
        baseBinding.textTitle.text = title
        baseBinding.buttonConfirm.setOnClickListener { onConfirm() }
        baseBinding.buttonCancel.setOnClickListener { onCancel() }
    }

    fun setButtonText(confirm: String?, cancel: String?): BaseDialog {
        baseBinding.buttonCancel.text = cancel
        baseBinding.buttonConfirm.text = confirm
        return this
    }

    protected fun setBottomVisible(visible: Boolean): BaseDialog {
        baseBinding.dialogBottom.visibility = if (visible) View.VISIBLE else View.GONE
        return this
    }

    protected fun setContent(view: View?): BaseDialog {
        baseBinding.contentLayout.addView(view)
        return this
    }

    open fun show() {
        dialog = AlertDialog.Builder(context)
                .setView(view)
                .create()
        val ad = dialog as AlertDialog?
        ad?.window?.setBackgroundDrawableResource(R.color.Transparent)
        ad?.show()
    }

    open fun onConfirm() {
        dialog?.dismiss()
    }

    open fun onCancel() {
        dialog?.dismiss()
    }
}